frontend/src/screens/Start.tsx

```diff
@@ -24,10 +24,31 @@ export default function Start() {
   const [unlockPassword, setUnlockPassword] = React.useState("");
   const [loading, setLoading] = React.useState(false);
   const [buttonText, setButtonText] = React.useState("Login");
-  useInfo(true); // poll the info endpoint to auto-redirect when app is running
+  const [startupError, setStartupError] = React.useState("");
+
+  const { data: info, isValidating } = useInfo(true); // poll the info endpoint to auto-redirect when app is running
 
   const { toast } = useToast();
 
+  React.useEffect(() => {
+    if (!isValidating && info?.startupError) {
```

Comment:
This isValidating check feels hacky. I don't think it's necessary?

Reply:
We need it actually, because we want to check everytime info is updated (and because we do it every 3 seconds and sometimes the error is not updated and info stays same, the useEffect hook is not invoked)

Reply:
> error is not updated

An example: suppose we try and fail (i.e. `info.startupError` is "connection error") on 1st second (assume info call is done on the same second, so the next will happen on 4s). Then we try again on 2s and as soon as we do the call, `SetStartupError(nil)` is called in the backend and usually it takes less than 3s for the call to fail, in that case the `info.startupError` would again be "connection error" (assuming the edge case where user changes nothing and tries again) and this would be set using `SetStartupError(err)` on 3s and when on 4s the info call is made, the info is the exact same although `info.startupError` changed from "connection error" to nil to "connection error" in between the 3s polling